clustered/server/src/main/java/org/ehcache/clustered/server/state/EhcacheStateContext.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ehcache.clustered.server.state;
+
+import com.tc.classloader.CommonComponent;
+
+/**
+ * Marker interface to pass context between begin and end message processing.
+ */
+@CommonComponent
```

Comment:
Neither AutoCloseable nor Closeable are annotated as such, why is this one special?

Reply:
Because `EhcacheStateContext` is implemented by a lambda. It implies that we expect it for be used as a functional interface (the annotation will enforce that). `Closeable` is not a functional interface. It is added to tell an object is closeable. But we expect the object to do other things.